File: LeetCode/1876. Substrings of Size Three with Distinct Characters/solution.py
class Solution:
    def countGoodSubstrings(self, s: str) -> int:
        count = i = 0
        while i + 2 < len(s):
            if (s[i + 1] == s[i + 2]):
                i += 2
                continue
            if (s[i] != s[i + 1] and s[i] != s[i + 2]):
                count += 1
            i += 1
        return count

File: LeetCode/1876. Substrings of Size Three with Distinct Characters/solution.java
class Solution {
  public int countGoodSubstrings(String s) {
      int count = 0, i = 0;
      while (i + 2 < s.length()) {
          if (s.charAt(i + 1) == s.charAt(i + 2)) {
              i += 2;
              continue;
          }
          if (s.charAt(i) != s.charAt(i + 1) && s.charAt(i) != s.charAt(i + 2)) {
              count++;
          }
          i++;
      }
      return count;
  }
}